docs/topics/composite-primary-key.txt

```diff
@@ -185,3 +185,13 @@ field :exc:`.FieldError`.
     This is also true of composite primary keys. Hence, you may want to set
     :attr:`.Field.editable` to ``False`` on all primary key fields to exclude
     them from ModelForms.
+
+Composite primary keys in model validation
+==========================================
+
+Since ``pk`` is only a virtual field, including ``pk`` as a field name in the
+``exclude`` argument of :meth:`.Model.clean_fields` has no effect. To exclude
```

Comment:
Is this consistent behaviour with other virtual fields? Would it be difficult to support virtual fields in `exclude`?

Reply:
Supporting was something I mentioned when raising the ticket, but Sarah helpfully pointed out that generated fields are also excluded from `exclude`. M2M fields are another story: they don't appear in `self._meta.fields`, so `clean_fields` never fathoms them anyway & wouldn't need to be excluded.